### ISSUE TITLE
Naming of batchnorm1d

### ISSUE THREAD

**Issue by nedtaylor** [nedtaylor/athena#11]
In PyTorch, batchnorm1d layers can either handle 2D or 3D data (where 2D input shape is (B, C) and 3D input shape is (B, C, L)).

B = batch size
C = number of channels
L = number of input features within the channel).

In our current implementation, ATHENA currently only handles 2D datatype.

I cannot currently see a way to make it handle both shapes (other than through the previously considered method of all data simply being 1D). As such, should there just be batchnorm0d and batchnorm1d?

https://pytorch.org/docs/stable/generated/torch.nn.BatchNorm1d.html

**Comment by nedtaylor** [comment]
This could likely just be handled by setting it to 1D if input shape is 1D, and 2D if input shape is 2D.

If only num_features (aka num_inputs) or num_channels is defined, then 1D data size is assumed. If both are defined, then it becomes 2D. If input shape is given as rank 1, then assume 1D input, if rank 2, then assume 2D input.

So, just use a 2D data shape all times. But just make one dimension have length of 1.

**Comment by nedtaylor** [comment]
Addressed in branch [batchnorm0d_1d](/nedtaylor/athena/tree/batchnorm0d_1d).
